alma/lib/Helpers/Admin/InsuranceHelper.php

```diff
@@ -216,4 +229,156 @@ public function saveConfigInsurance($config)
+          `id_product_attribute_insurance` int(10) unsigned NOT NULL,
+          `id_order` int(10) unsigned NULL,
+          `price` decimal(20,6) NOT NULL DEFAULT 0.000000,
+          `date_add` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+           PRIMARY KEY (`id_alma_insurance_product`)
+        ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci';
+
+        \Db::getInstance()->execute($sql);
+    }
+
+    /**
+     * Create the default Insurance product
+     * @return void
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    protected function createInsuranceProductIfNotExist()
+    {
+        $insuranceProduct = $this->productRepository->getProductIdByReference(
+            ConstantsHelper::ALMA_INSURANCE_PRODUCT_REFERENCE,
+            $this->context->language->id
+        );
+
+        if (!$insuranceProduct) {
+            try {
+                $product = new \Product();
+                $product->name = $this->createMultiLangField(utf8_encode(ConstantsHelper::ALMA_INSURANCE_PRODUCT_NAME));
+                $product->reference = ConstantsHelper::ALMA_INSURANCE_PRODUCT_REFERENCE;
+                $product->link_rewrite = $this->createMultiLangField(\Tools::str2url(ConstantsHelper::ALMA_INSURANCE_PRODUCT_NAME));
+                $product->id_category_default = ConstantsHelper::ALMA_INSURANCE_DEFAULT_CATEGORY;
+                $product->product_type = 'combinations';
+                $product->visibility = 'none';
+                $product->addToCategories(ConstantsHelper::ALMA_INSURANCE_DEFAULT_CATEGORY);
+                $product->add();
+
+                $shops = \Shop::getShops(true, null, true);
+                $image = new \Image();
+                $image->id_product = $product->id;
+                $image->cover = true;
+                if (($image->validateFields(false, true)) === true && ($image->validateFieldsLang(false, true)) === true && $image->add()) {
+                    $image->associateTo($shops);
+                    if (!$this->uploadImage($product->id, $image->id, ConstantsHelper::ALMA_INSURANCE_PRODUCT_IMAGE_URL)) {
+                        $image->delete();
+                    }
+                }
+            } catch (InsuranceInitException $e) {
+                Logger::instance()->error('[Alma] The insurance product has not been created');
+            }
+        }
+    }
+
+    /**
+     * Create the default Insurance attribute group
+     * @return void
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    protected function createAttributeGroupIfNotExist()
+    {
+        $insuranceAttributeGroup = $this->attributeGroupRepository->getAttributeIdByName(
+            ConstantsHelper::ALMA_INSURANCE_ATTRIBUTE_NAME,
+            $this->context->language->id
+        );
+
+        if (!$insuranceAttributeGroup) {
+            try {
+                $attributeGroup = new \AttributeGroup();
```

Comment:
move the code in try in a function create in a repository, don't forget to put comment